app/src/main/java/net/bible/service/db/readingplan/ReadingPlanDBAdapter.kt

```diff
@@ -0,0 +1,770 @@
+        val selection = "${ReadingPlanDays.COLUMN_READING_PLAN_META_ID}=? AND ${ReadingPlanDays.COLUMN_DAY_NUMBER}>?"
+        val selectionArgs = arrayOf(readingPlanMetaId.toString(), currentDay.toString())
+
+        val q = dbReadable.query(
+            ReadingPlanDays.TABLE_NAME,
+            projection,
+            selection,
+            selectionArgs,
+            null,
+            null,
+            ReadingPlanDays.COLUMN_DAY_NUMBER
+        )
+        var result = 0
+        if (q.moveToFirst()) {
+            result = q.getInt(0)
+            q.close()
+        }
+
+        // Update plan current day
+        if (result > 0) {
+            setMetaCurrentDayNumber(metaCurrentActiveReadingPlanID!!, result)
+        }
+
+
+        return result
+    }
+
+    fun setAllDaysReadUpTo(dayNumber: Int) {
+        val readingPlanMetaId: Int = metaCurrentActiveReadingPlanID!!
+        val numberOfReadings: Int = getDayNumberOfReadings(readingPlanMetaId,dayNumber)
+
+        val values = ContentValues().apply {
+            put(ReadingPlanDays.COLUMN_READ_STATUS,
+                ReadingPlanOneDayDB.ReadingStatus(
+                    readingPlanMetaId,
+                    numberOfReadings,
+                    getDayReadingChaptersArrayAsAllRead(numberOfReadings)
+                ).toJsonString()
+            )
+        }
+        val whereClause = "${ReadingPlanDays.COLUMN_READING_PLAN_META_ID}=? AND ${ReadingPlanDays.COLUMN_DAY_NUMBER}<=?"
+        val whereArgs = arrayOf(readingPlanMetaId.toString(), dayNumber.toString())
+
+        dbWritable.update(ReadingPlanDays.TABLE_NAME, values, whereClause, whereArgs)
+    }
+
+    fun getMetaIsDateBasedPlan(ReadingPlanMetaID: Int): Boolean {
+        return when (getMetaIntegerDB(ReadingPlanMetaID,ReadingPlanMeta.COLUMN_CURRENT_DAY)) {
+            ReadingPlanMeta.CONSTANT_CURRENT_DAY_BY_DATE -> true
+            else -> false
+        }
+    }
+
+    fun getMetaIsCustomUserAddedPlan(fileName: String): Boolean {
+        return when (getPlanDetailLink(null, fileName)) {
+            null -> true
+            else -> false
+        }
+    }
+
+    fun getMetaIsPlanAlreadyImported(fileName: String): Boolean {
+        val q = dbReadable.query(
+            ReadingPlanMeta.TABLE_NAME,
+            arrayOf(ReadingPlanMeta.COLUMN_ID),
+            "${ReadingPlanMeta.COLUMN_PLAN_FILE_NAME}=?",
+            arrayOf(fileName),
+            null,null,null)
+        var result = false
+        if (q.moveToFirst()) {
+            result = !q.isNull(0)
+        }
+        q.close()
+        return result
+    }
+
+    /**
+     * Get's Reading Plan description from resources strings. If not available there, it will try to get description
+     * from DB with second of this function with double constructor, here: [getMetaReadingPlanDescription]
+     */
+    fun getMetaReadingPlanDescription(ReadingPlanMetaID: Int): String {
+        return getPlanDetailLink(ReadingPlanMetaID)?.planDescription ?:
```

Comment:
weird name getPlanDetailLink -- what do you mean by 'link'?

Reply:
I guess it was just my way of easily distinguishing from DB data. With just PlanDetail it wouldn't be clear right away that this is not info from DB. With the link in the name, to me it's the link between DB and resource strings. But anyway, don't know if it's good or not.